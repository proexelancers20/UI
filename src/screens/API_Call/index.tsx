import React, {useEffect, useState} from 'react';
import {View, Text, StyleSheet, FlatList, Pressable} from 'react-native';
import colors from '../../helper/colors';

interface ListProps {
  body: string;
  id: number;
  title: string;
  userId: number;
}

const API_Call = () => {
  const [result, setResult] = useState<ListProps[]>([]);

  useEffect(() => {
    const callAPI = async () => {
      await fetch('https://jsonplaceholder.typicode.com/posts', {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
      }).then(async data => {
        let List = await data.json();
        if (List.length > 0) {
          setResult(List);
        }
      });
    };
    callAPI();
  }, []);

  const RenderRowItem = (itemName: string, item: string | number) => {
    function alert(item: string | number) {
      throw new Error('Function not implemented.');
    }

    return (
      <Pressable onPress={() => {alert(item)}}
        style={{
          flexDirection: 'row',
          marginBottom: 5,
        }}>
        <View style={{flex: 0.2}}>
          <Text>{itemName}</Text>
        </View>
        <View style={{flex: 0.8}}>
          <Text>{item}</Text>
        </View>
      </Pressable>
    );
  };

  const RenderItem = (item: ListProps) => {
    return (
      <View style={{padding: 15}}>
        {RenderRowItem('User Id :- ', item.userId)}
        {RenderRowItem('Title :- ', item.title)}
        {RenderRowItem('Body :- ', item.body)}
      </View>
    );
  };

  return (
    <View style={styles.container}>
      <FlatList
        testID='List_Data_FlatList'
        data={result}
        renderItem={({item}) => RenderItem(item)}
        initialNumToRender={20}
        keyExtractor={(_item, index) => index.toString()}
        ItemSeparatorComponent={() => {
          return <View style={styles.divide} />;
        }}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
  },
  divide: {height: 0.2, borderWidth: 0.2, borderColor: colors.black},
});

export default API_Call;
