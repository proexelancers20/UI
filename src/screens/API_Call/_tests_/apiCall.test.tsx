import React from 'react';
import API_Call from '..';
import {render, waitFor} from '@testing-library/react-native';

// Mock the fetch function
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve([
        {
          userId: 1,
          id: 1,
          title: 'Test Title 1',
          body: 'Test Body 1',
        },
        {
          userId: 2,
          id: 2,
          title: 'Test Title 2',
          body: 'Test Body 2',
        },
      ]), // Mocking an empty response for simplicity
  }),
);

const component = <API_Call />;

describe('api call screen', () => {
  test('snap', () => {
    const container = render(component);
    expect(container.toJSON()).toMatchSnapshot();
  });

  it('FlatList render list',async () => {
    const {getByTestId} = render(component);
    await waitFor(() => getByTestId('List_Data_FlatList'));
    const container = getByTestId('List_Data_FlatList');
    expect(container).toBeDefined()
    // expect(container.props.data.length).toBe(2);
  });
});