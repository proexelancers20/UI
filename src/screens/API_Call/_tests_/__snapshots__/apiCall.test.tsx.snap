// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api call screen snap 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    initialNumToRender={20}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="List_Data_FlatList"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
